src/main/java/core/basesyntax/storage/impl/StorageImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.storage.impl;
+
+import core.basesyntax.storage.Storage;
+import java.util.Collection;
+import java.util.HashMap;
+
+public class StorageImpl implements Storage {
+    private final HashMap<String, StorageEntry> entries;
```

Comment:
Consider using the Map interface as the type for the 'entries' field instead of a specific implementation like HashMap. This follows the principle of coding to an interface rather than an implementation, which promotes flexibility and scalability.